components/brave_wallet/browser/swap_request_helper.cc

```diff
@@ -132,4 +136,287 @@ std::optional<std::string> EncodeJupiterTransactionParams(
   return result;
 }
 
+}  // namespace jupiter
+
+namespace lifi {
+
+namespace {
+
+std::optional<std::string> EncodeChainId(const std::string& value) {
+  if (value == mojom::kSolanaMainnet) {
+    return "SOL";
+  }
+
+  uint256_t val;
+  if (!HexValueToUint256(value, &val)) {
+    return std::nullopt;
+  }
+
+  if (val > std::numeric_limits<uint64_t>::max() ||
+      val < std::numeric_limits<uint64_t>::min()) {
+    return std::nullopt;
+  }
+
+  return base::NumberToString(static_cast<uint64_t>(val));
+}
+
+base::Value::Dict EncodeToolDetails(
+    const mojom::LiFiToolDetailsPtr tool_details) {
```

Comment:
pass everywhere by ref if arg is not 'consumed'

Reply:
56d51daeb1738074c38dade4c96aec77d3863f8b